src/lexer.c

```diff
@@ -66,47 +66,84 @@ static bool IsString() {
 
   token.type = kTokenString;
 
+  token.precedence = kPrecPrimary;
+
   return true;
 }
 
 static bool IsCharacter() {
   switch (*source_code) {
     case '=':
-      token.type = kTokenEquals;
+      source_code++;
+      if (*source_code == '=') {
+        token.type = kTokenEquals;
+        token.precedence = kPrecComparison;
+        // NOLINTNEXTLINE(clang-analyzer-security.insecureAPI.DeprecatedOrUnsafeBufferHandling)
+        strncpy(token.value.text, "==", 3);
+        break;
+      }
+
+      token.type = kTokenAssign;
+      token.precedence = kPrecAssignment;
+      source_code--;
+
+      break;
+    case '!':
+      source_code++;
+      if (*source_code == '=') {
+        token.type = kTokenNotEquals;
+        token.precedence = kPrecComparison;
+        // NOLINTNEXTLINE(clang-analyzer-security.insecureAPI.DeprecatedOrUnsafeBufferHandling)
+        strncpy(token.value.text, "!=", 3);
+        break;
+      }
+      token.type = kTokenNot;
+      token.precedence = kPrecUnary;
+      source_code--;
+
       break;
     case '(':
       token.type = kTokenLeftParenthesis;
+      token.precedence = kPrecPrimary;
       break;
     case ')':
       token.type = kTokenRightParenthesis;
+      token.precedence = kPrecPrimary;
       break;
     case '+':
       token.type = kTokenPlus;
+      token.precedence = kPrecTerm;
       break;
     case '-':
       token.type = kTokenMinus;
+      token.precedence = kPrecTerm;
       break;
     case '*':
       token.type = kTokenStar;
+      token.precedence = kPrecFactor;
       break;
     case '/':
       token.type = kTokenSlash;
+      token.precedence = kPrecFactor;
       break;
     case ':':
       token.type = kTokenColon;
       break;
     case '>':
       source_code++;
-
       if ('=' == *source_code) {
         token.type = kTokenGreaterOrEquals;
+        token.precedence = kPrecComparison;
+        // NOLINTNEXTLINE(clang-analyzer-security.insecureAPI.DeprecatedOrUnsafeBufferHandling)
+        strncpy(token.value.text, ">=", 3);
 
         source_code++;
 
-        return true;
+        break;
```

Comment:
@WannabeMO This is incorrect. We don't want to break and jump to line 164. We want to return from the function here.

```suggestion
        return true;
```